src/server.ts

```diff
@@ -37,7 +37,7 @@ const ANNOUNCE_SCHEMA = z
 const WEBHOOK_SCHEMA = z
 	.object({
 		infoHash: z.string().length(40),
-		path: z.string().refine((path) => !path || existsSync(path)),
+		path: z.string().refine((path) => path && existsSync(path)),
 	})
 	.strict()
 	.partial()
```

Comment:
I think a union would be a truer representation of the actual data shape here and what we do with it. That would prevent the need to use Object.keys

https://zod.dev/?id=unions